nixtlats/data/datasets/m5.py

```diff
@@ -0,0 +1,137 @@
+        cache: bool
+            If `True` saves and loads
+
+        Notes
+        -----
+        [1] Returns train+test sets.
+        [2] Based on https://www.kaggle.com/lemuz90/m5-preprocess.
+        """
+        path = f'{directory}/m5/datasets'
+        file_cache = f'{path}/m5.p'
+
+        if os.path.exists(file_cache) and cache:
+            Y_df, X_df, S_df = pd.read_pickle(file_cache)
+
+            return Y_df, X_df, S_df
+
+        M5.download(directory)
+        # Calendar data
+        cal_dtypes = {
+            'd': 'category',
+            'wm_yr_wk': np.uint16,
+            'event_name_1': 'category',
+            'event_type_1': 'category',
+            'event_name_2': 'category',
+            'event_type_2': 'category',
+            'snap_CA': np.uint8,
+            'snap_TX': np.uint8,
+            'snap_WI': np.uint8,
+        }
+        cal = pd.read_csv(f'{path}/calendar.csv',
+                          dtype=cal_dtypes,
+                          usecols=list(cal_dtypes.keys()) + ['date'],
+                          parse_dates=['date'])
+
+        event_cols = [k for k in cal_dtypes if k.startswith('event')]
+        for col in event_cols:
+            cal[col] = cal[col].cat.add_categories('nan').fillna('nan')
+
+        # Prices
+        prices_dtypes = {
+            'store_id': 'category',
+            'item_id': 'category',
+            'wm_yr_wk': np.uint16,
+            'sell_price': np.float32
+        }
+
+        prices = pd.read_csv(f'{path}/sell_prices.csv',
+                             dtype=prices_dtypes)
+
+        # Sales
+        sales_dtypes = {
+            'id': 'category',
+            'item_id': prices.item_id.dtype,
+            'dept_id': 'category',
+            'cat_id': 'category',
+            'store_id': 'category',
+            'state_id': 'category',
+            **{f'd_{i+1}': np.float32 for i in range(1913)}
+        }
+        sales = pd.read_csv(f'{path}/sales_train_validation.csv',
+                            dtype=sales_dtypes)
+        long = sales.melt(id_vars=['id', 'item_id', 'dept_id', 'cat_id', 'store_id', 'state_id'],
+                          var_name='d', value_name='y')
+
+        long['d'] = long['d'].astype(cal.d.dtype)
+        long = long.merge(cal, on=['d'])
+        long = long.merge(prices, on=['store_id', 'item_id', 'wm_yr_wk'])
+        long = long.drop(columns=['d', 'wm_yr_wk'])
+
+        @njit
+        def first_nz_mask(x):
+            """Return a boolean mask where the True starts at the first non-zero value."""
+            mask = np.full(x.size, True)
+            for idx, value in enumerate(x):
+                if value == 0:
+                    mask[idx] = False
+                else:
+                    break
+            return mask
+
+        long = long.sort_values(['id', 'date'], ignore_index=True)
+        keep_mask = long.groupby('id')['y'].transform(lambda x: first_nz_mask(x.values))
+        long = long[keep_mask]
```

Comment:
```suggestion
        keep_mask = long.groupby('id')['y'].transform(first_nz_mask, engine='numba')
        long = long[keep_mask.astype(bool)]
```

Reply:
Solved in https://github.com/Nixtla/nixtlats/pull/63/commits/93a8f5490f54e7387d4964b997d92b5763ab545d